### ISSUE TITLE
CS3019 warning issued even when required to suppress CS3016

### ISSUE THREAD

**Issue by jkoritzinsky** [dotnet/roslyn#68560]
**Version Used**:  .NET 8.0.100 P4 SDK

**Steps to Reproduce**:

```csharp
using System;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

[assembly:CLSCompliant(true)]

[CLSCompliant(false)] // CS3019
internal class Baz
{
    [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvMemberFunction) })] // CS3016 if CLSCompliant(false) is removed
    internal static void Bar(nint x){}
}
```

Warnings:

```
warning CS3019: CLS compliance checking will not be performed on 'Baz' because it is not visible from outside this assembly
warning CS3016: Arrays as attribute arguments is not CLS-compliant
```

If the `CLSCompliant(false)` attribute is removed, then a the `CS3016` diagnostic is reported instead of `CS3019`.

[SharpLab link](https://sharplab.io/#v2:C4LgbgpgTglgZgTwD4AEBMBGAsAKBQBgAIUMA6AJQFcA7YGAWwlIGEB7egBxgBtoBlaGBgBjCAGcA3LgLEyVWgyYBJWtFYcBUIaMm5cAbQCGYsRHoAjbghDMAMnzaduMQ7QAUwKJQgBKALp6OHA8EMRohABirKy4AN64hImE+gCq1PSuhgDmEAAmzIbcvFBiAPLUVm4FRWzUYGKEALyE1BAA7vp+hLGEwAgcEKxwVYXctWAAsmbm0CgArHSs1D6EAL7+CUkwqlDUhbIAbMQALIQAQoZQbtTbwIQAHj6xq7gvOAZ2Duwczq7AbnBCqYNjhbtA9twwudDAAvOKbRKpdKZHL5UbQMoVBAjGpLepNFrtTrdXr9QbDapjPFTCyzBYwJYrdYBHBJQhg3b7EhHFCnC5XG60B5PN6rIA)

**Diagnostic Id**: CS3019

**Expected Behavior**:

No warning that CLS compilance checking will not be performed on a member when removing the attribute causes a CLS compliance warning.

**Actual Behavior**:

"warning CS3019: CLS compliance checking will not be performed on 'Baz' because it is not visible from outside this assembly" on `CLSCompliant(false)` when adding `CLSCompliant(false)` is suppressing a CS3016 warning.

See also: https://github.com/dotnet/roslyn/issues/68526

**Comment by elachlan** [comment]
Related: #4293
